rnd/autogpt_server/autogpt_server/blocks/ai.py

```diff
@@ -96,24 +92,24 @@ def parse_response(resp: str) -> tuple[dict[str, str], str | None]:
             except Exception as e:
                 return {}, f"JSON decode error: {e}"
 
-        prompt = [
-            {"role": "system", "content": trim_prompt(sys_prompt)},
-            {"role": "user", "content": trim_prompt(usr_prompt)},
-        ]
-
         logger.warning(f"LLM request: {prompt}")
         retry_prompt = ""
         for retry_count in range(input_data.retry):
             response_text = self.llm_call(
-                input_data.config.api_key,
-                input_data.config.model,
-                prompt
+                api_key=input_data.api_key.get(),
+                model=input_data.model,
+                prompt=prompt,
+                json=bool(input_data.expected_format),
             )
             logger.warning(f"LLM attempt-{retry_count} response: {response_text}")
 
-            parsed_dict, parsed_error = parse_response(response_text)
-            if not parsed_error:
-                yield "response", {k: str(v) for k, v in parsed_dict.items()}
+            if input_data.expected_format:
+                parsed_dict, parsed_error = parse_response(response_text)
+                if not parsed_error:
+                    yield "response", {k: str(v) for k, v in parsed_dict.items()}
+                    return
+            else:
+                yield "response", {"response": response_text}
```

Comment:
@Torantulino do want to consistently return a dict here? Or are we returning just string without the hard-coded `response` field, and change the output type to `dict[str, str] | str` ?